### PR TITLE
Nyc collapsible platform

### PR DESCRIPTION
**Pull request by dtch1997**

### CODE REVIEW

**Comment by dtch1997** [comment]
@yungchuenng great work! 

Based on my understanding of `GetFootDistToGround()`, the method is:
1) Get foot position using `getLinkState`
2) Draw a line downwards 5 units and check for collision using `rayTest` 

A couple quick questions: 
1) What units of distance are we using in pybullet simulation? (i.e. what does 5.0 mean)
2) I had a look at the documentation for `rayTest`. Regarding this code segment: 
```
      if data[0] == -1: # -1 if did not detect anything within detectable distance (out of range)
        distances[toe_link_index] = 5.0
```
Should we be using `data[1]` instead (i.e. link index)? Because that is documented to be -1 if no collision detected, while `data[0]` i.e. object unique ID is not documented to be -1 if no collision detected

File: blind_walking/envs/locomotion_gym_env.py [hunk 1]
```diff
@@ -162,8 +162,25 @@ def __init__(self,
         self.hf._generate_field(self,
                                 friction=self.height_field_friction,
                                 heightPerturbationRange=self.height_field_perturbation_range)
+    '''<<<<<<< HEAD
```
Comment (jennyzzt) [inline blind_walking/envs/locomotion_gym_env.py:165]: Is this merge conflict resolved?

File: blind_walking/robots/a1.py [hunk 2]
```diff
@@ -334,6 +335,23 @@ def GetFootForces(self):
...
+    distances = [0] * NUM_LEGS
+    for foot_id in self._foot_link_ids:
+      toe_link_index = self._foot_link_ids.index(
+        foot_id)
+      foot_position = self._pybullet_client.getLinkState(self.quadruped, foot_id)[0] # index 0 is linkWorldPosition
+      projection_position = copy.copy(foot_position)
+      projection_position[2] = projection_position[2] - detect_dist_from_foot_to_ground # always pointing downwards
+      data = self._pybullet_client.rayTest(foot_position, projection_position)
+      if data[0] == -1: # -1 if did not detect anything within detectable distance (out of range)
+        distances[toe_link_index] = 5.0
```
Comment (dtch1997) [inline blind_walking/robots/a1.py:350]: this looks like a bug, it should use `detect_dist_from_foot_to_ground` instead of being hardcoded

Comment (yungchuenng) [reply blind_walking/robots/a1.py:350]: Yes thats right! Thanks for checking!

**Comment by yungchuenng** [comment]
1. What units of distance are we using in pybullet simulation? (i.e. what does 5.0 mean)
- The unit is metres. 

2. I had a look at the documentation for rayTest. Regarding this code segment:
```
      if data[0] == -1: # -1 if did not detect anything within detectable distance (out of range)
        distances[toe_link_index] = 5.0
```
Should we be using data[1] instead (i.e. link index)? Because that is documented to be -1 if no collision detected, while data[0] i.e. object unique ID is not documented to be -1 if no collision detected. 

- I've tested and checked data[0] and data[1], for data[1], -1 also happens when it sees the ground (not just nothing within detectable distance). So using data[0] is more accurate such that when nothing is detected, then it'll just return the max. dist. 

Let me know if anything else!

**Comment by yungchuenng** [comment]
Hi Jenny, regarding the part in locomotion_gym_env.py, yes the conflict is resolved. The things in comment can be removed. I kept the original lines in master.

File: blind_walking/envs/locomotion_gym_env.py [hunk 3]
```diff
@@ -162,8 +162,25 @@ def __init__(self,
...
+                              texture='water_texture' # or 'grass_texture'
+                          )
+
+    # Generate collapsible platform or not
+    self.cp = CollapsiblePlatform()
+    if self.collapsible_platform:
+    ======='''
     # If collapsible tile or platform, enable hard reset
     elif self.collapsible_tile:
+      # >>>>>>> master
```
Comment (dtch1997) [inline blind_walking/envs/locomotion_gym_env.py:183]: Could we clean up the comments here from the merge conflict?

File: blind_walking/envs/sensors/robot_sensors.py [hunk 4]
```diff
@@ -149,6 +153,7 @@ def _get_observation(self) -> _ARRAY:
     else:
       motor_torques = self._robot.GetTrueMotorTorques()
     return motor_torques
+>>>>>>> master
```
Comment (dtch1997) [inline blind_walking/envs/sensors/robot_sensors.py:156]: Here as well

Comment (yungchuenng) [reply blind_walking/envs/sensors/robot_sensors.py:156]: got it

**Comment by dtch1997** [comment]
Looks good to me! Can be merged now

**Comment by yungchuenng** [comment]
Done :) Thanks for checking!

**Pull request merged**